src/Tizen.Account.OAuth2/Tizen.Account.OAuth2/Authorizer.cs

```diff
@@ -98,83 +103,98 @@ private TokenResponse GetRefreshToken(IntPtr requestHandle)
             IntPtr error = IntPtr.Zero;
-            TokenResponse response = null;
-            Interop.Manager.Oauth2RefreshTokenCallback accessTokenCb = (IntPtr responseHandle, IntPtr usrData) =>
+            IntPtr id = IntPtr.Zero;
+            lock (_taskResponseMap)
+            {
+                id = (IntPtr)_responseCompletionId++;
+            }
+            TaskCompletionSource<TokenResponse> tcsaccessTokenResponse = new TaskCompletionSource<TokenResponse>();
+            _taskResponseMap[id] = tcsaccessTokenResponse;
+            ret = Interop.Manager.RefreshAccessToken(_managerHandle, requestHandle, _accessTokenCb, id);
+             if (ret != (int)OAuth2Error.None || error != IntPtr.Zero)
             {
-                Interop.Response.GetError(responseHandle, out error);
                 if (error != IntPtr.Zero)
                 {
-                    Log.Error(ErrorFactory.LogTag, "Error occured");
+                    throw ErrorFactory.GetException(error);
                 }
                 else
                 {
-                    IntPtr accessToken;
-                    ret = Interop.Response.GetAccessToken(responseHandle, out accessToken);
-                    if (ret != (int)OAuth2Error.None)
-                    {
-                        Log.Error(ErrorFactory.LogTag, "Interop failed");
-                        throw ErrorFactory.GetException(ret);
-                    }
-
-                    IntPtr tokenType;
-                    ret = Interop.Response.GetTokenType(responseHandle, out tokenType);
-                    if (ret != (int)OAuth2Error.None)
-                    {
-                        Log.Error(ErrorFactory.LogTag, "Failed to get token type");
-                    }
-
-                    long expiresIn;
-                    ret = Interop.Response.GetExpiresIn(responseHandle, out expiresIn);
-                    if (ret != (int)OAuth2Error.None)
-                    {
-                        Log.Error(ErrorFactory.LogTag, "Failed to get expires in");
-                    }
-
-                    IntPtr refreshToken;
-                    ret = Interop.Response.GetRefreshToken(responseHandle, out refreshToken);
-                    if (ret != (int)OAuth2Error.None)
-                    {
-                        Log.Error(ErrorFactory.LogTag, "Interop failed");
-                        throw ErrorFactory.GetException(ret);
-                    }
-
-                    IntPtr scope;
-                    ret = Interop.Response.GetScope(responseHandle, out scope);
-                    if (ret != (int)OAuth2Error.None)
-                    {
-                        Log.Error(ErrorFactory.LogTag, "Failed to get scope");
-                    }
+                    Log.Error(ErrorFactory.LogTag, "Interop failed");
+                    throw ErrorFactory.GetException(ret);
+                }
+            }
+            tcsaccessTokenResponse.Task.ConfigureAwait(true);
```

Comment:
```suggestion
            tcsaccessTokenResponse.Task.ConfigureAwait(false);
```
Because no need to return to main loop

Reply:
implemented